src/pkg/services/m365/api/drive_pager_test.go

```diff
@@ -210,3 +214,226 @@ func (suite *DrivePagerIntgSuite) TestEnumerateDriveItems() {
 	require.NotEmpty(t, items, "should find items in user's drive")
 	assert.NotEmpty(t, du.URL, "should have a delta link")
 }
+
+// TestDriveDeltaPagerQueryParams is a regression test to check if
+// the delta pager is setting query params and prefer headers as
+// expected.
+func (suite *DrivePagerIntgSuite) TestDriveDeltaPagerQueryParams() {
+	tests := []struct {
+		name        string
+		setupf      func()
+		forURLCache bool
+		expect      assert.ErrorAssertionFunc
+	}{
+		// Separate out tests for $select and $prefer because otherwise
+		// it's hard to tell which gock check is failing.
+		{
+			name: "check $top",
+			setupf: func() {
+				delta := msDrive.NewItemItemsItemDeltaResponse()
+				delta.SetValue([]models.DriveItemable{
+					models.NewDriveItem(),
+				})
+
+				str := "deltaLink"
+				delta.SetOdataDeltaLink(&str)
+
+				interceptV1Path("drives", "drive", "items", "root", "delta()").
+					MatchParam("$top", strconv.Itoa(int(maxDeltaPageSize))).
+					Reply(200).
+					JSON(graphTD.ParseableToMap(suite.T(), delta))
+			},
+			expect: assert.NoError,
+		},
+		{
+			name: "check $select",
+			setupf: func() {
+				delta := msDrive.NewItemItemsItemDeltaResponse()
+				delta.SetValue([]models.DriveItemable{
+					models.NewDriveItem(),
+				})
+
+				str := "deltaLink"
+				delta.SetOdataDeltaLink(&str)
+
+				// This is a copy of DefaultDriveItemProps. This list should be
+				// adjusted while changing DefaultDriveItemProps. This however
+				// doesn't protect against adding new params to DefaultDriveItemProps
```

Comment:
tbh, having this separate from the set we configure in the code is probably a good thing. We need to start keeping better track of what params we're sending and when they're changed. Having a test failure should hopefully help a little bit with that